.htaccess

```diff
@@ -1,240 +1,561 @@
-    ExpiresByType text/x-javascript "access plus 1 month"
-    ExpiresByType application/x-shockwave-flash "access plus 1 month"
-    ExpiresByType image/x-icon "access plus 1 year"
-    ExpiresDefault "access plus 2 days"
+## If using SSL Offloading use the Offloader Header value set in 
+## Backend > System > Configuration > General > Secure. If the value is SSL_OFFLOADED
+## For HAProxy configuration: http-request add-header Ssl-Offloaded on
+## For Pound configuration: AddHeader "Ssl-Offloaded: on"
+
+	#SetEnvIf Ssl-Offloaded on HTTPS=on
+
+############################################
+## Uncomment if the Apache server is behind a load balancer to get the client's IPv4 address in the access log
+## Use the environment variable set bellow in the Apache server configuration
+## For example: LogFormat "%{HTTP_X_REMOTE_ADDR}e %l %u %t \"%r\" %>s %O \"%{Referer}i\" \"%{User-Agent}i\"" combined
+
+	#SetEnvIf X-Forwarded-For "^(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})$" HTTP_X_REMOTE_ADDR=$1
+
 </IfModule>
 
+<Files "cron.php">
+
 ############################################
-## By default allow all access
+## Enable cron access with base HTTP authorization
+## https://httpd.apache.org/docs/2.4/howto/auth.html
+##
+## Warning: .htpasswd file must be placed in a secure place to prevent outside access
+## If the document root is located in /usr/local/apache/htdocs
+## you might want to put the password file(s) in /usr/local/apache/
 
-    Order allow,deny
-    Allow from all
+	#AuthName "Cron Authorization"
+	#AuthUserFile ../.htpasswd
+	#AuthType basic
+	#Require valid-user
+
+############################################
+## The next methods must be comment if you enable cron access with base HTTP authorization
+##
+## Deny access (Default)
+## Comment the lines if you use the alternative method
+
+	Order allow,deny
+	Deny from all
 
 ###########################################
-## Deny access to release notes to prevent disclosure of the installed Magento version
+## Redirect to external location (Alternative)
+## Apache Module mod_rewrite must be enabled
 
-    <Files RELEASE_NOTES.txt>
-        order allow,deny
-        deny from all
-    </Files>
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
 
-############################################
-## If running in cluster environment, uncomment this
-## http://developer.yahoo.com/performance/rules.html#etags
+</Files>
+
+<FilesMatch "\.(all-contributorsrc|git.*/?)">
+	
+###########################################
+## Deny access (Default)
+## Comment the lines if you use the alternative method
 
-    #FileETag none
+	Order allow,deny
+	Deny from all
 
 ###########################################
-## Deny access to cron.php
-    <Files cron.php>
+## Redirect to external location (Alternative)
+## Apache Module mod_rewrite must be enabled
+
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
+
+</FilesMatch>
+
+<FilesMatch "(.*)\.(json|lock|md|sh|txt|sample)$">
```

Comment:
Is this blocking access to all json or txt files? I am sure that this would break part of someone's website if so.

Reply:
That directive locks all files with the respective extensions and which are found in OM root. If someone uploads such files in other directories they will be denied if they do not modify .htaccess files in those directories. If they were css, js, jpeg, php, html it was an issue, Those extensions don't seem to be very desirable to be opened in the browser.

Reply:
Maybe something like this (not tested):
```
<FilesMatch "^/[^/]*\.(json|lock|md|sh|txt|sample)$">
```

Or, just list out the files in the root dir you want to exclude, like `php.ini.sample`, `robots.txt.sample` etc.

I would be willing to bet that the blocking of `json` in particular will break some extension that tries to load a json file from the server in javascript.

Reply:
I will analyze the expression you proposed because I want to limit myself strictly to the root directory and only for those files. The rest of the files is controlled by .htaccess files that exist in most sub-directories. The idea of the deny was that in production no one would have to delete the files, but simply leave them there protected.